sbi/neural_nets/density_estimators/flow.py

```diff
@@ -58,6 +60,28 @@ def sample(self, sample_shape: Shape, condition: Tensor) -> Tensor:
         # nflows.sample() expects conditions to be batched.
         if len(condition.shape) == 1:
             condition = condition.unsqueeze(0)
-        return self.net.sample(num_samples, context=condition).reshape(
-            (*sample_shape, -1)
+            batch_dim = 1
+        else:
+            batch_dim = condition.shape[0]
+        samples = self.net.sample(num_samples, context=condition).reshape(
+            (batch_dim, *sample_shape, -1)
         )
+        return samples.squeeze(0)
```

Comment:
Why the squeezing? Because we want to return without batch dim if `batch_dim=1?`

Reply:
I am confused about the `batch_dim` now. Given `num_samples = n` we should return something like `(n, ...)`, no? 
But now we are returning something like `(b, n, ...)`. 
Maybe I am missing something.

Reply:
I see, this is what `nflows` does before return samples: 
```python
samples = torchutils.split_leading_dim(samples, shape=[-1, num_samples])
```

I guess the idea is that when passed multiple different conditions, i.e., `c.shape = (M, x_shape)`, then it will return `(M, num_samples, x_shape)`.

Reply:
Feel free to resolve this if you think what I said makes sense :)

Reply:
Yeah, exactly. The squeezing is because we have to unsqueeze before (nflows assert). 

But this will also happen if we input an `(1, x_shape)` vector, so I guess it would be better to then also return `(1, num_samples, theta_shape)`. (I will make a proper if else for `context.ndim=1` and `context.ndim>1`.